### ISSUE TITLE
[dev-libs/libvterm] Live ebuild depends on 'bzr' eclass, which doesn't exist anymore in Portage

### ISSUE THREAD

**Issue by Dr-Terrible** [leycec/raiagent#90]
emerge then fails with:

```sh
ERROR: dev-libs/libvterm-9999-r2::raiagent failed (depend phase):
 *   bzr.eclass could not be found by inherit()
 *
 * Call stack:
 *                 ebuild.sh, line 609:  Called source '/usr/portage/overlays/raiagent/dev-libs/libvterm/libvterm-9999-r2.ebuild'
 *   libvterm-9999-r2.ebuild, line   6:  Called inherit 'multilib' 'bzr' 'flag-o-matic'
 *                 ebuild.sh, line 290:  Called die
 * The specific snippet of code:
 *              [[ -z ${location} ]] && die "${1}.eclass could not be found by inherit()"
 *
 * If you need support, post the output of `emerge --info '=dev-libs/libvterm-9999-r2::raiagent'`,
 * the complete build log and the output of `emerge -pqv '=dev-libs/libvterm-9999-r2::raiagent'`.
 * Working directory: '/usr/lib/python3.7/site-packages'
 * S: '/var/tmp/portage/dev-libs/libvterm-9999-r2/work/libvterm-9999'
```

**Comment by leycec** [comment]
Thanks for the heads up, inappropriately named @Dr-Terrible! It makes sense that `bzr.eclass` is now dead, given the general death of every DVCS not named `git`. I'll get this patched up in a relative jiffy.

Also, **thanks so much for striving for reproducibility, repeatability, and correctness in science.** My wife and I no longer work in academia for the inverse reason; in our several decades worth of experience, most academics in our fields (regenerative medicine and computational biology) actively strive for the opposite – which is probably why the [replication crisis](https://en.wikipedia.org/wiki/Replication_crisis) still remains a thing.

Ultimately, what academia needs are hard real-world consequences for fraudulent scientific behaviour. I'd settle for merely double-blinding the review process coupled with publicizing reviewer names, but tenured professors fight tooth and nail against even those reasonable advances. To paraphrase the Vonnegut, "So it goes."

**Comment by leycec** [comment]
**Back to Gentoo Land.** I sadly have no idea how to resolve this to everyone's satisfaction. I'd love to preserve live `libvterm` support, but:

* [`bzr.eclass` has been officially last-rited.](https://bugs.gentoo.org/726300) Technically, we could get around that by just importing the most recent `bzr.eclass` commit from the Portage tree into this overlay – except...
* [`dev-vcs/bzr` has *also* been officially last-rited.](https://www.mail-archive.com/gentoo-dev@lists.gentoo.org/msg88757.html) This makes sense, because Bazaar has been effectively dead since 2017 and remains tied-at-the-hip to Python 2, which is also now effectively dead. Technically, we could get around that by *also* importing the most recent `dev-vcs/bzr` commit from the Portage tree into this overlay – except that [`python:2` is also going away shortly](https://gentoo.org/support/news-items/2020-02-07-python-2-7-eol.html), so that would be cray-cray.

My other thought was to just refactor our live `libvterm` ebuild to fetch tarball snapshots of the most recent `libvterm` commits (i.e., `HEAD`) from the existing Bazaar repo – except Portage sensibly requires tarballs to be explicitly checksummed and I would have no reasonable way of doing that.

In short, there are no good solutions. The onus lies on [Paul Evans](http://www.leonerd.org.uk/contact) to host his codebases with a sane, well-maintained DVCS. `git` is the obvious choice, but I'd settle for Mercurial. Until then, I'm reluctantly last-riting `libvterm` from our overlay. *It is a sad day.* :sob:
